Makefile

```diff
@@ -114,521 +105,312 @@ endef
+$(addsuffix .force-symbols,$(notdir $(wildcard $(BUILD_DIR:$(VERSION)=us)/*.elf))): %.elf.force-symbols: | $(VENV_DIR)
+	$(PYTHON) tools/symbols.py elf $(BUILD_DIR)/$*.elf > $(CONFIG_DIR)/symbols$(if $(filter-out stmad,$*),.us).$*.txt
+force-symbols: $(addsuffix .force-symbols,$(notdir $(wildcard $(BUILD_DIR:$(VERSION)=us)/*.elf)))
+
+context: $(M2CTX_APP) | $(VENV_DIR)
+ifndef SOURCE
+	$(error SOURCE environment variable must be set to generate context)
+endif
+	VERSION=$(VERSION) $(PYTHON) $(M2CTX_APP) $(SOURCE)
+	$(call echo,ctx.c has been updated.)
+
+disks/us: | $(SOTNDISK)
+	$(SOTNDISK) extract disks/sotn.$(VERSION).cue $(EXTRACTED_DISK_DIR)
+disks/pspeu:
+	mkdir -p $(EXTRACTED_DISK_DIR)
+	7z x -y disks/sotn.pspeu.iso -o$(EXTRACTED_DISK_DIR)
+disks/saturn:
+	bchunk disks/sotn.$(VERSION).bin disks/sotn.$(VERSION).cue disks/sotn.$(VERSION).iso
+	7z x disks/sotn.$(VERSION).iso01.iso -o$(EXTRACTED_DISK_DIR) || true
+extract-disk: $(EXTRACTED_DISK_DIR)
+
+ovl_to_bin  = $(call to_upper,$(call add_ovl_prefix,$(1),ST/,BOSS/)/$(1).BIN $(call add_ovl_prefix,$(1),ST/,BOSS/)/F_$(1).BIN)
+DISK_PREPARE_FILES := DRA.BIN BIN/RIC.BIN ST/SEL/SEL.BIN $(addprefix SERVANT/,$(call to_upper,$(addsuffix .BIN,$(SERVANTS))))
+DISK_PREPARE_FILES += $(foreach target,$(filter-out sel,$(STAGES)) $(BOSSES),$(call ovl_to_bin,$(target)))
+disk-prepare: build $(SOTNDISK)
+	mkdir -p $(BUILD_DISK_DIR); cp -r $(EXTRACTED_DISK_DIR)/* $(BUILD_DISK_DIR)
+	cp $(BUILD_DIR)/main.exe $(BUILD_DISK_DIR)/SLUS_000.67
+	$(call echo,cp $(BUILD_DIR)/*.BIN $(BUILD_DISK_DIR)/) $(foreach item,$(DISK_PREPARE_FILES),cp $(BUILD_DIR)/$(notdir $(item)) $(BUILD_DISK_DIR)/$(item);)
+disk-debug: disk-prepare
+	cd tools/sotn-debugmodule && $(MAKE)
+	cp $(BUILD_DIR:$(VERSION)=)/sotn-debugmodule.bin $(BUILD_DISK_DIR)/SERVANT/TT_000.BIN
+	$(SOTNDISK) make $(BUILD_DIR:$(VERSION)=)/sotn.$(VERSION).cue $(BUILD_DISK_DIR) $(CONFIG_DIR)/disk.$(VERSION).lba
+disk: disk-prepare
+	$(SOTNDISK) make $(BUILD_DIR:/$(VERSION)=)/sotn.$(VERSION).cue $(BUILD_DISK_DIR) $(CONFIG_DIR)/disk.$(VERSION).lba
+
+# Targets for copying the physical disk to an image file
+$(addprefix disks/,sotn.%.bin sotn.%.cue): PHONY
+	$(muffle)( which -s cdrdao && which -s toc2cue ) || (echo "cdrdao(1) and toc2cue(1) must be installed" && exit 1 )
+	$(call echo,Dumping disk)
+	$(muffle)cd disks && \
+        DEVICE="$(shell cdrdao scanbus 2>&1 | grep -vi cdrdao | head -n1 | sed 's/ : [^:]*$$//g')" && \
+        cdrdao read-cd \
+            --read-raw \
+            --datafile sotn.$*.bin \
+            --device "$$DEVICE" \
+            --driver generic-mmc-raw \
+            sotn.$*.toc && \
+        toc2cue sotn.$*.toc sotn.$*.cue && \
+        rm sotn.$*.toc
+$(addprefix dump-disk,.eu .hk .jp10 .jp11 .saturn .us .usproto): disks/sotn.$(VERSION).cue
+dump-disk.%: PHONY# Last resort
+	$(error Automated dumping of $* is not supported)
+dump-disk: dump-disk.$(VERSION)
```

Comment:
I made no functional changes here, only reworked the target logic so that it is more version agnostic and will return the same message for any version that is not matched by a specific target.